dbt_coves/utils/airbyte_api.py

```diff
@@ -0,0 +1,114 @@
+from typing import Dict
+from requests.exceptions import RequestException
+import requests, json
+
+
+class AirbyteApiCallerException(Exception):
+    pass
+
+
+class AirbyteApiCaller:
+    def api_call(self, endpoint: str, body: Dict[str, str] = None):
+
+        """
+        Generic `api caller` for contacting Airbyte
+        """
+        try:
+            response = requests.post(endpoint, json=body)
+            if response.status_code == 204:
+                return response
+            elif response.status_code >= 200 and response.status_code < 300:
+                return json.loads(response.text)
+            else:
+                raise RequestException(
+                    f"Unexpected status code from airbyte: {response.status_code}"
+                )
+        except RequestException as e:
+            raise AirbyteApiCallerException("Airbyte API error: " + str(e))
+
+    def __init__(self, api_host, api_port):
+        try:
+            airbyte_host = api_host
+            airbyte_port = api_port
+            airbyte_api_root = "api/v1/"
+            airbyte_api_base_endpoint = (
+                f"http://{airbyte_host}:{airbyte_port}/{airbyte_api_root}"
+            )
+        except ValueError as e:
```

Comment:
I don't see how a `ValueError` exception could be raised

Reply:
You need to push changes so I can see them. I still don't understand why you need to catch a general Exception in simple variables assigments.

Reply:
Removed the try-except

7f7dcc2[](url)